src/backends/fvm.hpp

```diff
@@ -2,6 +2,6 @@
 
 #include "fvm_multicore.hpp"
 
-#ifdef WITH_CUDA
+#ifdef NMC_HAVE_CUDA
```

Comment:
Again, how about `NMC_WITH_CUDA`, otherwise we will be getting confused and wondering when to use `HAVE` and when to use `WITH`.